lib/std/os.zig

```diff
@@ -4526,7 +4526,7 @@ pub const UnexpectedError = error{
 
 /// Call this when you made a syscall or something that sets errno
 /// and you get an unexpected error.
-pub fn unexpectedErrno(err: usize) UnexpectedError {
+pub fn unexpectedErrno(err: c_int) UnexpectedError {
```

Comment:
I'm not sure this is the way to go. If not linking libc, `os.getErrno()` returns a `u12` not a `c_int` on linux at least. Do we really want to tie this API to the C integer type?

Reply:
I am not very familiar with linux kernel internals. I assume the errno size used by kernel is part of its ABI.
But at least `u12` fully fits inside a `c_int` (assuming at least 16 bits)

Reply:
Yes; however we don't want to use larger types just because C did...